drivers/st25r3911b/st25r3911b_spi.c

```diff
@@ -0,0 +1,298 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <zephyr.h>
+
+#include <spi.h>
+#include <device.h>
+#include <soc.h>
+#include <gpio.h>
+#include <logging/log.h>
+
+#include "st25r3911b_spi.h"
+#include "st25r3911b_reg.h"
+
+LOG_MODULE_REGISTER(st25r3911b, CONFIG_ST25R3911B_LIB_LOG_LEVEL);
+
+static struct device *spi_dev;
+
+#define ST25R3911B_READ_REG(_reg) (0x40 | (_reg))
+#define ST25R3911B_WRITE_REG(_reg) (~0xC0 & (_reg))
+#define ST25R3911B_DIRECT_CMD(_cmd) (0xC0 | (_cmd))
+#define ST25R3911B_FIFO_READ 0xBF
+#define ST25R3911B_FIFO_WRITE 0x80
+
+#define REG_CNT 0x3F
+
+#define _DO_SPI_PORT_NUM(_num) \
+	 DT_SPI_ ## _num ## _NAME
+
+#define SPI_PORT_NUM(_num) \
+	_DO_SPI_PORT_NUM(_num)
+
+#define _DO_CS_PORT_NUM(_spi_port) \
+	DT_NORDIC_NRF_SPI_ ## _spi_port ## _CS_GPIOS_CONTROLLER
```

Comment:
`DT_NORDIC_NRF_SPI_SPI_`, otherwise there is no guarantee you reference the proper SPI instance.

Reply:
Ok